### PR TITLE
Validator Vesting Upgrade v44 + Cosmos x/auth migration

### PR DESCRIPTION
**Pull request by DracoLi**
- Upgrades validator-vesting to v44. Since all validator vesting accounts should be migrated to a base account now, I removed most of validator-vesting logic aside from client stuff since we use it for supply related queries.
- Cosmos x/auth migration can now be added now that validator vesting is migrated to v44. I've included that + some test cases to make sure the accounts are properly migrated.

The `appstate-auth-v15.json` contains both auth, supply, and bank app states since v44 moved supply and auth coins to the bank module. The tests here should cover those and it seems to be correctly migrated.

### CODE REVIEW

File: x/validator-vesting/client/cli/query.go [hunk 1]
```diff
@@ -5,212 +5,237 @@ import (
...
-func queryCirculatingSupply(queryRoute string, cdc *codec.Codec) *cobra.Command {
+func queryCirculatingSupply() *cobra.Command {
 	return &cobra.Command{
 		Use:   "circulating-supply",
 		Short: "Get circulating supply",
 		Long:  "Get the current circulating supply of kava tokens",
 		Args:  cobra.NoArgs,
 		RunE: func(cmd *cobra.Command, args []string) error {
-			cliCtx := context.NewCLIContext().WithCodec(cdc)
+			cliCtx, err := client.GetClientTxContext(cmd)
```
Comment (drklee3) [inline x/validator-vesting/client/cli/query.go:47]:
```suggestion
			cliCtx, err := client.GetClientQueryContext(cmd)
```

I switched these up a lot before too, same for the other query commands

File: migrate/v0_16/testdata/appstate-bank-v15.json [hunk 2]
```diff
@@ -0,0 +1,541 @@
...
+          "account_number": "0",
+          "address": "kava1wuzhkn2f8nqe2aprnwt3jkjvvr9m7dlkpumtz2",
+          "coins": [
+            {
+              "amount": "1000000000000",
+              "denom": "ukava"
+            }
+          ],
+          "public_key": null,
+          "sequence": "0"
```
Comment (rhuairahrighairidh) [inline migrate/v0_16/testdata/appstate-bank-v15.json:482]: nit: all the `sequence` values are 0 which can hide bugs. If the migrations got the tag name wrong they'd look for a nonexistent field, and fill in the value with the default 0. I'd maybe change a couple to something other than 0.

File: migrate/v0_16/testdata/appstate-bank-v15.json [hunk 3]
```diff
@@ -0,0 +1,541 @@
...
+        "value": {
+          "account_number": "0",
+          "address": "kava1ypjp0m04pyp73hwgtc0dgkx0e9rrydecm054da",
+          "coins": [
+            {
+              "amount": "1000000000000",
+              "denom": "ukava"
+            }
+          ],
+          "public_key": null,
```
Comment (rhuairahrighairidh) [inline migrate/v0_16/testdata/appstate-bank-v15.json:78]: nit: All the public_key's are empty, I'd add some to test.

**Review by rhuairahrighairidh — APPROVED** [review]
There's a rogue print statement, but otherwise looks good.

I think a test running on a full mainnet export would be good to sanity check account balances and no panics. But that can be done later, I added a card.

File: migrate/v0_16/testdata/appstate-bank-v15.json [hunk 4]
```diff
@@ -0,0 +1,541 @@
...
+              "amount": "1000000000000"
+            },
+            {
+              "denom": "hard",
+              "amount": "1000000000000"
+            },
+            {
+              "denom": "swp",
+              "amount": "1000000000000"
+            }
```
[the remark this answers is not in the document]
Comment (DracoLi) [reply migrate/v0_16/testdata/appstate-bank-v15.json:20]: 👍

**Comment by DracoLi** [comment]
@rhuairahrighairidh thx for the review. addressed feedbacks here https://github.com/Kava-Labs/kava/pull/1072/commits/7a124ad76861fb0105834458db1412c9dbbb50fb

**Pull request merged**